digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/lane/LaneFiller.scala

```diff
@@ -0,0 +1,422 @@
+    lanes.sortBy(s => 0L-s.modifiedDateTime.getOrElse(s.createdDateTime.getOrElse(DateTime.now())).getMillis)
+  }
+
+  private def dropShortSegments(roadLink: RoadLink, lanes: Seq[PersistedLane], changeSet: ChangeSet): (Seq[PersistedLane], ChangeSet) = {
+    val (shortSegments, linearSegments) = lanes.partition(a => (a.endMeasure - a.startMeasure) < MinAllowedLength && roadLink.length >= MinAllowedLength )
+    val expiredLaneIds = shortSegments.map(_.id).toSet
+
+    (linearSegments, changeSet.copy(expiredLaneIds = changeSet.expiredLaneIds ++ expiredLaneIds.filterNot(_ == 0L)))
+  }
+
+  private def adjustAssets(roadLink: RoadLink, lanes: Seq[PersistedLane], changeSet: ChangeSet): (Seq[PersistedLane], ChangeSet) = {
+    lanes.foldLeft((Seq[PersistedLane](), changeSet)){
+      case ((resultAssets, change), lane) =>
+        val (asset, adjustmentsMValues) = adjustAsset(lane, roadLink)
+        (resultAssets ++ Seq(asset), change.copy(adjustedMValues = change.adjustedMValues ++ adjustmentsMValues))
+    }
+  }
+
+  private def adjustAsset(lane: PersistedLane, roadLink: RoadLink): (PersistedLane, Seq[MValueAdjustment]) = {
+    val roadLinkLength = GeometryUtils.geometryLength(roadLink.geometry)
+    val adjustedStartMeasure = if (lane.startMeasure < AllowedTolerance && lane.startMeasure > MaxAllowedError) Some(0.0) else None
+    val endMeasureDifference: Double = roadLinkLength - lane.endMeasure
+    val adjustedEndMeasure = if (endMeasureDifference < AllowedTolerance && endMeasureDifference > MaxAllowedError) Some(roadLinkLength) else None
+    val mValueAdjustments = (adjustedStartMeasure, adjustedEndMeasure) match {
+      case (None, None) => Nil
+      case (s, e)       => Seq(MValueAdjustment(lane.id, lane.linkId, s.getOrElse(lane.startMeasure), e.getOrElse(lane.endMeasure)))
+    }
+    val adjustedAsset = lane.copy(
+      startMeasure = adjustedStartMeasure.getOrElse(lane.startMeasure),
+      endMeasure = adjustedEndMeasure.getOrElse(lane.endMeasure))
+    (adjustedAsset, mValueAdjustments)
+  }
+
+
+  private def adjustLanesSideCodes(roadLink: RoadLink, lanes: Seq[PersistedLane], changeSet: ChangeSet): (Seq[PersistedLane], ChangeSet) = {
+
+    // auxiliary function to create PersistedLane object
+    def createPersistedLane(laneCode: Int, sideCode: Int, municipalityCode: Long, baseProperties: Seq[LaneProperty] ): PersistedLane = {
+      val lanePropertiesValues = LanePropertiesValues(baseProperties ++ Seq(LaneProperty("lane_code", Seq(LanePropertyValue(laneCode)))))
+
+      PersistedLane(0L, roadLink.linkId, sideCode, laneCode, municipalityCode,
+        0, roadLink.length, None, None, None, None, expired = false, roadLink.vvhTimeStamp, None,
+        lanePropertiesValues)
+    }
+
+
+    if(lanes.isEmpty )
+      return (lanes, changeSet)
+
+    val lanesToProcess = lanes.filter(_.linkId == roadLink.linkId)
+    val baseLane = lanesToProcess.sortBy(_.laneCode).head
```

Comment:
```scala
val baseLane = lanesToProcess.minBy(_.laneCode)

Reply:
Done.

Reply:
However this code is the same as AssetFiller